api/app/Models/Pool.php

```diff
@@ -632,43 +636,45 @@ public function scopeOrderByPoolBookmarks(Builder $query, ?array $args): Builder
         return $query;
     }
 
-    public function scopeAuthorizedToView(Builder $query)
+    public function scopeAuthorizedToView(Builder $query): void
     {
-
         /** @var \App\Models\User */
         $user = Auth::user();
 
-        if (! $user) {
-            return $query->where('published_at', '<=', Carbon::now()->toDateTimeString());
+        // can view any pool - return query with no filters added
+        if ($user?->isAbleTo('view-any-pool')) {
+            return;
         }
 
-        if (! $user->isAbleTo('view-any-pool')) {
-            $query->where(function (Builder $query) use ($user) {
-
-                if ($user->isAbleTo('view-team-draftPool')) {
-                    // Only add teams the user can view pools in to the query for `whereHas`
-                    $teams = $user->rolesTeams()->get();
-                    $teamIds = [];
-                    foreach ($teams as $team) {
-                        if ($user->isAbleTo('view-team-draftPool', $team)) {
-                            $teamIds[] = $team->id;
-                        }
+        // we might want to add some filters for some pools
+        $filterCountBefore = count($query->getQuery()->wheres);
+        $query->where(function (Builder $query) use ($user) {
+            if ($user?->isAbleTo('view-team-draftPool')) {
+                // Only add teams the user can view pools in to the query for `whereHas`
+                $teams = $user->rolesTeams()->get();
+                $teamIds = [];
+                foreach ($teams as $team) {
+                    if ($user->isAbleTo('view-team-draftPool', $team)) {
+                        $teamIds[] = $team->id;
                     }
-
-                    $query->orWhereHas('legacyTeam', function (Builder $query) use ($teamIds) {
-                        return $query->whereIn('id', $teamIds);
-                    });
                 }
 
-                if ($user->isAbleTo('view-any-publishedPool')) {
-                    $query->orWhere('published_at', '<=', Carbon::now()->toDateTimeString());
-                }
+                $query->orWhereHas('legacyTeam', function (Builder $query) use ($teamIds) {
+                    return $query->whereIn('id', $teamIds);
+                });
+            }
 
-                return $query;
-            });
+            if ($user?->isAbleTo('view-any-publishedPool')) {
+                $query->orWhere('published_at', '<=', Carbon::now()->toDateTimeString());
+            }
+        });
+        $filterCountAfter = count($query->getQuery()->wheres);
+        if ($filterCountAfter > $filterCountBefore) {
```

Comment:
Hmm. Won't this always be true, because of the `where` at line 651?

Reply:
I guess not! I tested it in tinker. And `->where( function ($q) { ... } )` with an empty inner function doesn't actually increase the where count. Could you add a little comment mentioning that this has been considered and isn't a problem?

Reply:
[extra comments](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11138/commits/dbda192c713a308da0628b1501ca454a706f879a)